### PR TITLE
[now dev] Special case `@now/static-build` build trigger

### PR DESCRIPTION
**Pull request by AndyBitz**
Special case `@now/static-build` build trigger, since it will never build without a restart otherwise

### CODE REVIEW

File: src/util/dev/server.ts [hunk 1]
```diff
@@ -239,6 +239,34 @@ export default class DevServer {
         }
       }
     }
+
+    // Special case `@now/static-build`,
+    // since it won't trigger builds otherwise
+    ((nowConfig && nowConfig.builds) || []).map((build) => {
```
Comment (styfle) [inline src/util/dev/server.ts]:
```suggestion
    ((nowConfig && nowConfig.builds) || []).forEach((build) => {
```

It looks like the returned value from map is not used so probably use a for each instead

File: src/util/dev/server.ts [hunk 2]
```diff
@@ -423,17 +451,17 @@ export default class DevServer {
...
     // We need to delete these properties for zero config to work
     // with file changes
     if (this.cachedNowConfig) {
       delete this.cachedNowConfig.builds;
       delete this.cachedNowConfig.routes;
     }
 
+    // The default empty `now.json` is used to serve all files as static
+    // when no `now.json` is present
+    let config: NowConfig = this.cachedNowConfig || { version: 2 };
```
Comment (styfle) [inline src/util/dev/server.ts:463]: Why was this moved down?

Comment (AndyBitz) [reply src/util/dev/server.ts:463]: @styfle I moved the `delete` statements up, to make it more clear that when `this.cachedNowConfig` is used the `builds` and `routes` property are not available anymore

**Review by styfle — APPROVED** [review]

**Comment by AndyBitz** [comment]
Already fixed by #2562
